tests/tests.nix

```diff
@@ -146,6 +132,53 @@ in
     ];
   };
 
+  including-a-file-also-includes-the-parents = rec {
+    root = ./fixture1;
+    actual = nix-filter {
+      inherit root;
+      include = [
+        "src/innerdir/inner.js"
+      ];
+    };
+    expected = [
+      "src"
+      "src/innerdir"
+      "src/innerdir/inner.js"
+    ];
+  };
+
+  including-a-directory-also-includes-the-parents = rec {
+    root = ./fixture1;
+    actual = nix-filter {
+      inherit root;
+      include = [
+        "src/components"
+      ];
+      exclude = [
+        (nix-filter.matchExt "jsx")
+      ];
+    };
+    expected = [
+      "src"
+      "src/components"
+    ];
+  };
+
+  including-a-directory-also-includes-the-childs = rec {
+    root = ./fixture1;
+    actual = nix-filter {
+      inherit root;
+      include = [
+        "src/innerdir"
+      ];
+    };
+    expected = [
+      "src"
+      "src/innerdir"
+      "src/innerdir/inner.js"
+    ];
+  };
```

Comment:
Can you also test the filter on the exclude? I think the new logic would fail on something like this:

```
nix-filter {
   inherit root;
   include = [
      "src/innerdir/one.js"
   ];
   exclude = [
     "src/innerdir/two.js"
   ];
};
```
because the exclude will match the folder and exclude it.